src/components/settings/PushNotifications/PushNotificationsBanner/index.tsx

```diff
@@ -87,21 +87,23 @@ export const _getSafesToRegister = (
 }
 
 export const PushNotificationsBanner = ({ children }: { children: ReactElement }): ReactElement => {
-  const isNotificationsEnabled = useHasFeature(FEATURES.PUSH_NOTIFICATIONS)
+  const isNotificationFeatureEnabled = useHasFeature(FEATURES.PUSH_NOTIFICATIONS)
   const chain = useCurrentChain()
   const totalAddedSafes = useAppSelector(selectTotalAdded)
   const { safe, safeAddress } = useSafeInfo()
   const addedSafesOnChain = useAppSelector((state) => selectAddedSafes(state, safe.chainId))
   const { query } = useRouter()
   const onboard = useOnboard()
 
+  const { getPreferences, getAllPreferences } = useNotificationPreferences()
   const { dismissPushNotificationBanner, isPushNotificationBannerDismissed } = useDismissPushNotificationsBanner()
 
   const isSafeAdded = !!addedSafesOnChain?.[safeAddress]
-  const shouldShowBanner = isNotificationsEnabled && !isPushNotificationBannerDismissed && isSafeAdded
+  const isSafeRegistered = getPreferences(safe.chainId, safeAddress)
```

Comment:
Should we write a unittest for this component?

Reply:
I've added tests covering the logic related to this PR. I'll add another test in the other open one as well.